RSHighscores.body.php

```diff
@@ -69,6 +69,42 @@ private static function retrieveHiscores( $hs, $player ) {
 	}
 
 	/**
+	 * Lookup hiscores data from object cache before retrieving from the site.
+	 *
+	 * @param string $hs Which hiscores API to retrieve from.
+	 * @param string $player Player's display name.
+	 * @return string Raw hiscores data
+	 */
+	private static function lookupHiscores( $hs, $player ) {
+		// Instance of the object cache to retrieve the hiscores data from.
+		$objCache = wfGetCache( CACHE_ANYTHING );
+
+		// Try to retrieve the hiscores data from the object cache.
+		$data = $objCache->get( 'rshiscores-' . $player . '-' . $hs );
+
+		// Couldn't find in the object cache, so retrieve from the site.
+		if ( $data === false ) {
+			// If not blocked from too many requests or technical issues, then lookup.
+			if ( $objCache->get( 'rshiscores-blocked') === false ) {
+				$data = self::retrieveHiscores( $hs, $player );
+
+				// Request failed, so no requests for 15 min.
+				if ( $data == 'C28' ) {
```

Comment:
This is definitely what's returned when we get blacklisted, but I'm not sure if there is something intermediate, e.g. being rate limited. I know HTML pages give you a message, but I've no idea about the APIs. ASking around some of the other fansites, one guy seems to think you get a 404, but I'm slightly sceptical as he demonstrated it using a URL that would always be a 404 (something that doesn't exist and never will).